src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs

```diff
@@ -52,10 +54,23 @@ public static IServiceCollection AddApplicationInsights(this IServiceCollection
                 });
 
             services.AddSingleton<ITelemetryInitializer, HttpDependenciesParsingTelemetryInitializer>();
+            services.AddSingleton<ITelemetryInitializer>(provider =>
+            {
+                var httpContextAccessor = provider.GetService<IHttpContextAccessor>();
```

Comment:
If `EnableHttpTriggerExtendedInfoCollection` is false, can we skip this initializer as well?